src/prefect_server/api/runs.py

```diff
@@ -473,14 +477,29 @@ async def get_runs_in_queue(
 
         run_labels = flow_run.labels
 
+        # Agent label filtering
         # if the run labels are a superset of the provided labels, skip
         if set(run_labels) - set(labels):
             continue
 
+        # Agent label filtering
         # if the run has no labels but labels were provided, skip
         if not run_labels and labels:
             continue
 
+        # Concurrency label filtering
+        # Concurrency limiting final check is in state setting pipeline
+        if run_labels:
```

Comment:
Not locking on concurrency checks since this is supposed to be a more liberal filter. The only purpose is to prevent sending agents flow runs that we already know there are 0 available slots for, not to do fancy counting math that sends _only_ the correct number of flow runs (although we can do it, it just increases complexity).